hibernate-core/src/main/java/org/hibernate/context/internal/ThreadLocalSessionContext.java

```diff
@@ -111,7 +111,7 @@ private boolean needsWrapping(Session session) {
 		// try to make sure we don't wrap and already wrapped session
 		if ( Proxy.isProxyClass( session.getClass() ) ) {
 			final InvocationHandler invocationHandler = Proxy.getInvocationHandler( session );
-			if ( invocationHandler != null && TransactionProtectionWrapper.class.isInstance( invocationHandler ) ) {
+			if ( TransactionProtectionWrapper.class.isInstance( invocationHandler ) ) {
```

Comment:
We don't need nullness checking for either `instanceof` or `Class#isInstance()` will take care of that already